R/get-pkgs.R

```diff
@@ -7,13 +7,9 @@ deepstate_get_mismatched_datatypes <- function(params.list){
   params.list <-gsub(" ","",params.list)
   datatypes <- list("NumericVector","NumericMatrix" ,"arma::mat","double",
                   "string","CharacterVector","int","IntegerVector")
```

Comment:
this is the list of supported data types right? So rather than "mismatched" in the function name maybe "unsupported" would be more appropriate?
Also this list is redundant with the logic in https://github.com/FabrizioSandri/RcppDeepState/blob/master/R/fun_harness_create.R -- if you have time it would be good to reorganize that code so that a list or table is used to lookup what to do, rather than the current code which is rather difficult to understand (lots of if/else statements).

Reply:
That's true, this list is redundant. I completely reviewed the `deepstate_fun_create` function inside [fun_harness_create.R](https://github.com/FabrizioSandri/RcppDeepState/blob/master/R/fun_harness_create.R) and replaced the if else statements with a table containing information for the supported datatypes. In detail each column of this table corresponds to a supported datatype and provides some details on it. The first row of this table contains an alternative datatype to be used when saving the inputs(e.g., using `qs::c_qsave`), whereas the second row correspond to the associated generation function that takes as input a range. When a datatype contains a value of NA in both rows, it means that is supported, utilizes itself when executing `qs::c_qsave` and lacks a range function. In this way checking if a datatype is supported is really simple: just check if exists a column for the datatype in this table.